src/rpc/blockchain.cpp

```diff
@@ -164,6 +164,7 @@ UniValue blockheaderToJSON(const CBlockIndex& tip, const CBlockIndex& blockindex
     result.pushKV("mediantime", blockindex.GetMedianTimePast());
     result.pushKV("nonce", blockindex.nNonce);
     result.pushKV("bits", strprintf("%08x", blockindex.nBits));
+    result.pushKV("target", GetTarget(tip, pow_limit).GetHex());
```

Comment:
In 2a7bfebd5e788e1d9e7e07a9f1b8e3625a0301cd _Add target to getblock(header) in RPC and REST_: as pointed out in #33440 this is a bug. Unlike `getmininginfo` in baa504fdfaff4a9f61bc939035df5d5f2978cfd7 and `getblockchaininfo` in f153f57acc9f9a6f84af161d5bed9aa9965abaa3, we should not use the `tip` here but `blockindex` (like for nBits).